course-project/Backend/src/db/models/records_test.go

```diff
@@ -0,0 +1,252 @@
+			}
+		})
+	}
+}
+
+func TestUpdate(t *testing.T) {
+	t.Parallel()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// given
+	mockPool := pgxpoolmock.NewMockPgxPool(ctrl)
+	res := pgconn.CommandTag{byte(100)}
+	mockPool.EXPECT().Exec(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(res, nil)
+
+	mockPool2 := pgxpoolmock.NewMockPgxPool(ctrl)
+	res2 := pgconn.CommandTag{byte(0)}
+	mockPool2.EXPECT().Exec(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(res2, pgx.ErrNoRows)
+
+	type args struct {
+		conn db.DBConn
+		id   uint64
+		rec  Record
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{name: "first", args: args{conn: mockPool}, wantErr: false},
+		{name: "error", args: args{conn: mockPool2}, wantErr: true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := Update(tt.args.conn, tt.args.id, tt.args.rec); (err != nil) != tt.wantErr {
+				t.Errorf("Update() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestDelete(t *testing.T) {
+	t.Parallel()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// given
+	mockPool := pgxpoolmock.NewMockPgxPool(ctrl)
+	res := pgconn.CommandTag{byte(100)}
+	mockPool.EXPECT().Exec(gomock.Any(), gomock.Any(), gomock.Any()).Return(res, nil)
+
+	mockPool2 := pgxpoolmock.NewMockPgxPool(ctrl)
+	res2 := pgconn.CommandTag{byte(0)}
+	mockPool2.EXPECT().Exec(gomock.Any(), gomock.Any(), gomock.Any()).Return(res2, pgx.ErrNoRows)
+
+	type args struct {
+		conn db.DBConn
+		id   uint64
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{name: "first", args: args{conn: mockPool}, wantErr: false},
+		{name: "error", args: args{conn: mockPool2}, wantErr: true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := Delete(tt.args.conn, tt.args.id); (err != nil) != tt.wantErr {
+				t.Errorf("Delete() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestInsert(t *testing.T) {
+	t.Parallel()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	// given
+	mockPool := pgxpoolmock.NewMockPgxPool(ctrl)
+	res := mockRow{
+		id:            1,
+		cityName:      "Arkham",
+		timeRequested: "mock time",
+		temperature:   0.0,
+	}
+	mockPool.EXPECT().QueryRow(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(res)
+
+	mockPool2 := pgxpoolmock.NewMockPgxPool(ctrl)
+	res2 := mockRow{
+		id:            10,
+		cityName:      "City of Sin",
+		timeRequested: "666",
+		temperature:   10000.0,
+	}
+
+	mockPool2.EXPECT().QueryRow(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(res2)
+
+	type args struct {
+		conn db.DBConn
+		rec  Record
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    uint64
+		wantErr bool
+	}{
+		{name: "first", args: args{conn: mockPool, rec: Record{
```

Comment:
This test case fails with a panic.

Reply:
Fixed: had wrong number of place holders for Query() in mock pgx pool